app/routes/surveys/components/Submissions/Results.js

```diff
@@ -65,6 +84,36 @@ const Results = ({
     }
   ];
 
+  const switchGraph = (id, index) => {
+    const newQuestions = survey.questions;
+    const questionToUpdate = newQuestions.find(question => question.id === id);
+    if (questionToUpdate) {
+      questionToUpdate.displayType =
+        questionToUpdate.displayType === 'pie_chart'
+          ? 'bar_chart'
+          : 'pie_chart';
+    }
+    const qIndex = newQuestions.indexOf(
```

Comment:
You can move this `qIndex` inside the `if (questionToUpdate) {`, so you won't need the same `if` twice.